readthedocs/restapi/views/footer_views.py

```diff
@@ -101,13 +101,18 @@ def footer_html(request):
 
     version_compare_data = get_version_compare_data(project, version)
 
+    domain = project.subdomain()
+    versions = project.ordered_active_versions(user=request.user)
+    for each in versions:
+        each.subdomain_url = each.get_subdomain_url(domain)
```

Comment:
This is a clever way of moving the query from the template to the python code, and speeding it up.

Reply:
I believe we are on the right track, but as written, this does not yet reduce the number of queries. In my testing of this PR, I'm still seeing 4x queries per call to `get_subdomain_url`.

A good way to see this yourself is to add a logger for `django.db` into `readthedocs.settings.base.py` (you also need to change the `console` level to `DEBUG`:

            'django.db': {
                'handlers': ['console'],
                'level': 'DEBUG',
                'propagate': False,
            },

Then add some logging right before this line. There should not be any queries between calls.

Reply:
I got the point why it is still making queries, while we are not calling `resolve _domain` in `resolve` function but we are still making calls to `resolve_path` and that function is making those queries(for getting private and cname  which is `None` in various cases. I will soon fix those issues by passing some more `**kwargs`.